src/json-api-adapter.js

```diff
@@ -10,44 +10,79 @@ DS._routes = Ember.create(null);
 
 DS.JsonApiAdapter = DS.RESTAdapter.extend({
   defaultSerializer: 'DS/jsonApi',
+
+  getSelfRoute: function(typeName, id, record) {
+    if(!this.serializer || !this.serializer.buildSelfKey) { return null; }
+    var route;
+    if(record) {
+      route = DS._routes[this.serializer.buildSelfKey(record.type, record.id, typeName, id)]
+        || DS._routes[this.serializer.buildSelfKey(record.type, null, typeName, null)];
+      if(route) { return route; }
+    }
+    return DS._routes[this.serializer.buildSelfKey(null, null, typeName, id)]
+      || DS._routes[this.serializer.buildSelfKey(null, null, typeName, null)];
+  },
+
+  getResourceRoute: function(typeName, id) {
+    var routeName = (id) ? typeName + '.' + id : typeName;
+    return DS._routes[routeName] || DS._routes[typeName];
+  },
+
+  getRoute: function(typeName, id, record) {
+    var route;
+    if(record) {
+      route = this.getSelfRoute(typeName, id, record);
+      if(route) { return route; }
+    }
+    return this.getResourceRoute(typeName, id);
```

Comment:
hm, could you explain this method to me? I'm not that deeply into this topic. But data must be only loaded from a `resource` url if it is related data to something and the IDs for that or the object itself was not previously loaded inside the `linked` field.